cEP-0001.md

```diff
@@ -4,38 +4,101 @@ Decision Making in coala
+official coala team member and can use cobot to label issues and pull requests.
+Developers also have voting rights on A level decisions.
+
+If the contributions of a developer show the lack of fundamental understanding
+of the coala workflow they can be demoted to the newcomer rank with a B level
+decision. The promotion to developer must be granted after a renewed completion
+of the newcomer guide.
+A developer can choose to be demoted newcomer at any time.
+
+### Maintainer
 
-### Members
+A Maintainer is someone that showed persistence in contributions and or an
+increased quality compared to a developer. They offer experience and passion for
+the project and are granted full voting rights due to their achievements.
+
+A developer can be promoted to a maintainer with a B level decision. This decision
+does not have to be initiated by a maintainer so it is valid for a developer to
+propose himself for a promotion.
+A maintainer has all the rights a developer has. Additionally maintainers have
+write access to coala repositories, can promote newcomers and developers and
+have voting rights on B and C level decisions.
+
+The two important aspects of the maintainer rank are the voting rights for B and
+C level decisions and the write access. While the write access mostly is a
+convenience feature that is granted to long time contributors as a quality of
+life improvement and sign of trust, the voting rights also show the valuation
+of ones knowledge and opinions.
+It is due to these considerations, that it is valid to promote contributors to
+maintainers that did not finish the newcomer guide. This enables non coding
+contributors to add their input to important decisions.
+
+The demotion from maintainer to developer represents a C level decision. It
+should only be done if a maintainer has shown inactivity for prolonged periods
+of time, half a year and upwards.
+__This should not be used as a tool to silence unwanted opinions!__
+Maintainers may choose to be demoted to developers at any time.
 
-Contributors may be promoted to members with a level B decision. Members have
-normal voting rights as described below.
 
 ### Benevolent Limited Dictator (BLD)
 
 Members of the coala community have suggested the establishment of a Benevolent
-Limited Dictator (short: BLD).
+Limited Dictator (short: BLD) similar to the Python community's Guido.
+
+The BLD has the same rights as a maintainer with additional veto rights as
+described below. As per this cEP, Lasse Schuirmann is the BLD for coala.
+
+The BLD can not be demoted at any point but may decide to resign and be demoted
+to a maintainer any time.
+
+### Removal from the coala community
+
+The removal of any member from the coala community represents a C level
+decision. This includes loss of all ranks and rights in the community and
+declares that person unwanted as a member of the coala community.
+
+This status stays active unlimited but can be revoked at any point by a C level
```

Comment:
`indefinitely` might be a better word choice than `unlimited` here 👍 after that should finally be done